recipes-ni/ni-base-system-image-tests/files/fs_permissions_diff.py

```diff
@@ -109,110 +105,208 @@ def upload_manifest(db, fs_manifest, logger):
     data['fs_permissions'] = header + fs_manifest
 
     record = db.insert(data)
-    logger.log('INFO: Uploaded fs permissions of OS {} from {} with _id {}'.format(data['os_version_full'], data['date'], record.inserted_id))
+    logger.log('INFO: Uploaded fs permissions of OS {} {} from {} with _id {}'.format(data['os_version_codename'], data['os_version_full'], data['date'], record.inserted_id))
 
 def strip_headers(fs_manifest):
     return ''.join(line + '\n' for line in fs_manifest.splitlines() if not line.startswith('#'))
 
-def get_old_fs_manifest(db, logger):
+def get_old_fs_manifests(db, logger):
+    def run_query(query):
+        nonlocal db
+        nonlocal logger
+
+        sort_order = [('os_version', pymongo.DESCENDING), ('date', pymongo.DESCENDING)]
+
+        if db.count_documents(query):
+            results = db.find(query).sort(sort_order).limit(1)
+        else:
+            # If there are no results, there may not be a run from the current os version.
+            # Remove that requirement.
+            del query['os_version_codename']
+            logger.log('INFO: No prior fs permissions from this OS codename found. Relaxing that constraint.')
+
+            if db.count_documents(query):
+                results = db.find(query).sort(sort_order).limit(1)
+            else:
+                logger.log('INFO: No suitable previous fs permissions found')
+                return '', '<none>'
+
+        result = next(results)
+
+        logger.log('INFO: Using previous fs permissions of OS {} {} from {} with _id {}'.format(result['os_version_codename'], result['os_version_full'], result['date'], result['_id']))
```

Comment:
Could we replace the output generated from this line with something similar to this:

basis   = 23.0.0.f147 from 1/27/2023 with _id = 432987123
recent  = 23.3.0.d102 from 2/27/2023 with _id = 432987234
current = 23.3.0.d103 from 3/1/2023 with _id = 432987235

Maybe place this output as close as possible to the start of the diffs so the reader has more context about what "basis/recent/current" mean in the diff output.